brigade/core/task.py

```diff
@@ -67,19 +69,22 @@ class Result(object):
         diff (obj): Diff between state of the system before/after running this task
         result (obj): Result of the task execution, see task's documentation for details
         host (:obj:`brigade.core.inventory.Host`): Reference to the host that lead ot this result
+        skipped (bool): ``True`` if the host was skipped because there was a previous error
 
     Attributes:
         changed (bool): ``True`` if the task is changing the system
         diff (obj): Diff between state of the system before/after running this task
         result (obj): Result of the task execution, see task's documentation for details
         host (:obj:`brigade.core.inventory.Host`): Reference to the host that lead ot this result
+        skipped (bool): ``True`` if the host was skipped because there was a previous error
     """
 
-    def __init__(self, host, result=None, changed=False, diff="", **kwargs):
+    def __init__(self, host, result=None, changed=False, diff="", skipped=False, **kwargs):
```

Comment:
I am wondering if it'd be interesting to not assume we skipped due to an error and add a "reason" as well.

Reply:
What do you mean by a reason? Do you have any example of what that might be.

Reply:
Right now we are assuming `skipped` can only be because a task failed so we took it from the pool. There might be other reasons in the future so we could have something like:

```
class Skipped(object):
    def __init__(self, reason=None):
        self.reason = reason  # probably a set of standard reasons?
```

and then use it as:

```
return Result(skipped=Skipped("errored"))
```

so the user can do:

```
if result.skipped.reason == "error":
    # oh well
```